File: src/main/java/me/kuku/yuq/service/impl/WeiboServiceImpl.java
package me.kuku.yuq.service.impl;

import com.icecreamqaq.yudb.jpa.annotation.Transactional;
import me.kuku.yuq.dao.WeiboDao;
import me.kuku.yuq.entity.WeiboEntity;
import me.kuku.yuq.service.WeiboService;

import javax.inject.Inject;
import java.util.List;

public class WeiboServiceImpl implements WeiboService {
    @Inject
    private WeiboDao weiboDao;
    @Override
    @Transactional
    public WeiboEntity findByQQ(Long qq) {
        return weiboDao.findByQQ(qq);
    }

    @Override
    @Transactional
    public void save(WeiboEntity weiboEntity) {
        weiboDao.saveOrUpdate(weiboEntity);
    }

    @Override
    @Transactional
    public int delByQQ(Long qq) {
        return weiboDao.delByQQ(qq);
    }

    @Override
    @Transactional
    public List<WeiboEntity> findByMonitor(Boolean monitor) {
        return weiboDao.findByMonitor(monitor);
    }

    @Override
    @Transactional
    public List<WeiboEntity> findAll() {
        return weiboDao.findAll();
    }
}

File: src/main/kotlin/me/kuku/yuq/service/impl/WeiboServiceImpl.kt
package me.kuku.yuq.service.impl

import com.icecreamqaq.yudb.jpa.annotation.Transactional
import me.kuku.yuq.dao.WeiboDao
import me.kuku.yuq.entity.WeiboEntity
import me.kuku.yuq.service.WeiboService
import javax.inject.Inject

class WeiboServiceImpl: WeiboService {
    @Inject
    private lateinit var weiboDao: WeiboDao

    @Transactional
    override fun findByQQ(qq: Long) = weiboDao.findByQQ(qq)

    @Transactional
    override fun save(weiboEntity: WeiboEntity) = weiboDao.saveOrUpdate(weiboEntity)

    @Transactional
    override fun delByQQ(qq: Long) = weiboDao.delByQQ(qq)
}